### PR TITLE
fix for assisted reference type arguments of factories

### LINKED ISSUE #147: Factories with reference types broken in fruit 3.7.0
In version 3.6 a factory with `fruit::Assisted<T&>` types could be used.
In version 3.7.0 existing code does not compile anymore.

### PR DESCRIPTION
**Pull request by tiolan**
fixes #147

### CODE REVIEW

**Comment by poletti-marco** [comment]
Thanks for the PR. Added a minor comment about the test, other than that (as long as ci tests pass) this LGTM.

File: tests/test_register_factory.py [hunk 1]
```diff
@@ -389,6 +389,38 @@ def test_autoinject_2_assisted_params(self):
             COMMON_DEFINITIONS,
             source)
 
+    def test_autoinject_assisted_params_ref(self):
```
[the remark this answers is not in the document]
Comment (tiolan) [reply tests/test_register_factory.py:470]: Valid point. I added two additional tests for assisted reference types.
- One for move-only type
- One for non-move, non-copy type
I also left this test case, as it makes sure different types (rref, ref, value) can be used simultaneously.
In combination with the other tests we know, parameters will not be copied.
If you want me to remove this test though, I am happy to do so.

**Comment by tiolan** [comment]
> Oh and 1 job in the windows CI workflow is known to be failing: https://github.com/google/fruit/actions/runs/3815971015 I need to look into it. So you can ignore that failure.

After the latest change all GH actions in my fork ran through, except the Windows one.

**Comment by tiolan** [comment]
Will it be possible to do a 3.7.1 release in case this is taken?
If so, what would be the process? Would that need to be part of this PR?

**Comment by poletti-marco** [comment]
> Will it be possible to do a 3.7.1 release in case this is taken?
> If so, what would be the process? Would that need to be part of this PR?

I can do that once this PR is in.
I need to fix the windows CI first, I'll do the fix and then the release this weekend.

**Comment by poletti-marco** [comment]
All LGTM now, thanks for adding the tests.
I kicked off a CI run on this PR and I'll merge it once that's complete.

**Comment by poletti-marco** [comment]
Thanks. Merging.

**Pull request merged**